From 398471d96cfea0424c2adae1630f085d8bba0986 Mon Sep 17 00:00:00 2001
From: Samuel Verschelde <stormi-xcp@ylix.fr>
Date: Fri, 25 Aug 2023 17:47:34 +0200
Subject: [PATCH 19/19] Install /etc/multipath/conf.d/custom.conf

Update Makefile so that the file is installed along with sm.

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index b0ae353..3357cbf 100755
--- a/Makefile
+++ b/Makefile
@@ -78,6 +78,7 @@ SM_LIBS += sr_health_check
 UDEV_RULES = 65-multipath 55-xs-mpath-scsidev 57-usb 58-xapi
 MPATH_DAEMON = sm-multipath
 MPATH_CONF = multipath.conf
+MPATH_CUSTOM_CONF = custom.conf
 SMLOG_CONF = SMlog
 
 SM_XML := XE_SR_ERRORCODES
@@ -93,6 +94,7 @@ UDEV_SCRIPTS_DIR := /etc/udev/scripts/
 SYSTEMD_SERVICE_DIR := /usr/lib/systemd/system/
 INIT_DIR := /etc/rc.d/init.d/
 MPATH_CONF_DIR := /etc/multipath.xenserver/
+MPATH_CUSTOM_CONF_DIR := /etc/multipath/conf.d/
 MODPROBE_DIR := /etc/modprobe.d/
 EXTENSION_SCRIPT_DEST := /etc/xapi.d/extensions/
 LOGROTATE_DIR := /etc/logrotate.d/
@@ -146,6 +148,7 @@ install: precheck
 	mkdir -p $(SM_STAGING)$(INIT_DIR)
 	mkdir -p $(SM_STAGING)$(SYSTEMD_SERVICE_DIR)
 	mkdir -p $(SM_STAGING)$(MPATH_CONF_DIR)
+	mkdir -p $(SM_STAGING)$(MPATH_CUSTOM_CONF_DIR)
 	mkdir -p $(SM_STAGING)$(MODPROBE_DIR)
 	mkdir -p $(SM_STAGING)$(LOGROTATE_DIR)
 	mkdir -p $(SM_STAGING)$(DEBUG_DEST)
@@ -163,6 +166,8 @@ install: precheck
 	  $(SM_STAGING)$(SM_DEST)/plugins/
 	install -m 644 multipath/$(MPATH_CONF) \
 	  $(SM_STAGING)/$(MPATH_CONF_DIR)
+	install -m 644 multipath/$(MPATH_CUSTOM_CONF) \
+	  $(SM_STAGING)/$(MPATH_CUSTOM_CONF_DIR)
 	install -m 755 multipath/sm-multipath \
 	  $(SM_STAGING)/$(INIT_DIR)
 	install -m 755 multipath/multipath-root-setup \
-- 
2.41.0

